pgmpy/estimators/base.py

```diff
@@ -1,5 +1,9 @@
 #!/usr/bin/env python
 from warnings import warn
+try:
+    from functools import lru_cache
```

Comment:
@ankurankan functools lru_cache is as fast the attached lru_cache. Actually I think implementation is same if you checkout the lru_cache implementation in functools.

Reply:
@khalibartan Yeah, but that might change in newer versions of python. Plus things might also break in `backports.lru_cache` for newer versions of python. And I think it's better to always use the default implementation if possible because python community will be fixing any new issues/bugs whereas we can't have that guarantee for third party modules.